fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1158,18 +1162,208 @@ void partitionedJoinedIndexTest() {
 
             String luceneSearch = "simple_text: \"fathers\"";
-            QueryComponent filter = new LuceneQueryComponent(luceneSearch, List.of("simple", "complex"));
-            RecordQuery query = RecordQuery.newBuilder()
-                    .setRecordType("luceneJoinedPartitionedIdx")
-                    .setFilter(filter)
-                    .setRequiredResults(List.of(Key.Expressions.field("simple").nest("text")))
-                    .build();
-            final RecordQueryPlan plan = planner.plan(query);
-            final List<?> results = plan.execute(recordStore).asList().join();
-            assertNotNull(results);
-            assertEquals(1, results.size());
+
+            LuceneScanQueryParameters scan = new LuceneScanQueryParameters(
+                    ScanComparisons.EMPTY,
+                    new LuceneQueryMultiFieldSearchClause(LuceneQueryType.QUERY, luceneSearch, false),
+                    new Sort(new SortField("complex.timestamp", SortField.Type.LONG, true)),
+                    List.of("complex.timestamp"),
+                    List.of(LuceneIndexExpressions.DocumentFieldType.LONG),
+                    null);
+            LuceneScanQuery scanQuery = scan.bind(recordStore, JOINED_INDEX_NOGROUP, EvaluationContext.EMPTY);
+            RecordCursor<IndexEntry> cursor = recordStore.scanIndex(JOINED_INDEX_NOGROUP, scanQuery, null, ExecuteProperties.newBuilder().setReturnedRowLimit(15).build().asScanProperties(false));
+            List<IndexEntry> entries = cursor.asList().join();
+            assertNotNull(entries);
+            assertEquals(1, entries.size());
+            commit(context);
+        }
+    }
+
+    private enum SortType {
+        ASCENDING,
+        DESCENDING,
+        UNSORTED,
+    }
+
+    static Stream<Arguments> continuationDuringRepartitioningTest() {
+        return Stream.of(
+                // grouped, synthetic
+                Arguments.of(true, true, SortType.ASCENDING),
+                Arguments.of(true, true, SortType.DESCENDING),
+                Arguments.of(true, true, SortType.UNSORTED),
+                // ungrouped, synthetic
+                Arguments.of(false, true, SortType.ASCENDING),
+                Arguments.of(false, true, SortType.DESCENDING),
+                Arguments.of(false, true, SortType.UNSORTED),
+                // grouped, simple
+                Arguments.of(true, false, SortType.ASCENDING),
+                Arguments.of(true, false, SortType.DESCENDING),
+                Arguments.of(true, false, SortType.UNSORTED),
+                // ungrouped, simple
+                Arguments.of(false, false, SortType.ASCENDING),
+                Arguments.of(false, false, SortType.DESCENDING),
+                Arguments.of(false, false, SortType.UNSORTED)
+        );
+    }
+
+    @ParameterizedTest(name = "isGrouped: {0}, isSynthetic: {1}, sort type: {2}")
+    @MethodSource
+    void continuationDuringRepartitioningTest(boolean isGrouped,
```

Comment:
As a followup it would probably be good to also add a search to the randomized test. It won't have as strict assertions, but it could assert that the two sorted end up getting all the ids, and the unsorted doesn't throw an error. It could create a couple queries with random limits before partitioning, and then continue to get the remainder.